webrtc/RTCTransport-helper.js

```diff
@@ -0,0 +1,126 @@
+'use strict';
+
+/*
+  Code using this helper should also include RTCPeerConnection-helper.js
+  in the main HTML file
+
+  The following helper functions are called from RTCPeerConnection-helper.js:
+    getTrackFromUserMedia
+    addTrackOrTransceiver
+    exchangeIceCandidates
+    doSignalingHandshake
+*/
+
+// Wait for all DTLS and ICE transport of a peer connection
+// to turn to connected or completed state. When the RTCPeerConnectionState
+// is "connected", all RTCIceTransports and RTCDtlsTransports are in the
+// "connected", "completed" or "closed" state and at least one of them is in the
+// "connected" or "completed" state.
+async function waitForConnectedState(pc) {
+  if (pc.connectionState === 'connected') {
+    return;
+  }
+
+  return new Promise((resolve, reject) => {
+    pc.addEventListener('connectionstatechange', () => {
+      const { connectionState } = pc;
+
+      if (connectionState === 'connected') {
+        resolve();
+      } else if (['closed', 'failed'].includes(connectionState)) {
+        reject(new Error(`one of DTLS/ICE transports transition to unexpected state ${connectionState}`));
+      }
+    });
+  });
+}
+
+function getDtlsTransportFromSctpTransport(sctpTransport) {
+  assert_true(sctpTransport instanceof RTCSctpTransport,
+    'Expect pc.sctp to be instantiated from RTCSctpTransport');
+
+  const dtlsTransport = sctpTransport.transport;
+  assert_true(dtlsTransport instanceof RTCDtlsTransport,
+    'Expect sctp.transport to be an RTCDtlsTransport');
+
+  return dtlsTransport;
```

Comment:
I confess I'm personally not a fan of tests hidden in helpers and helpers that otherwise merely substitute JS for English—I prefer helpers that encapsulate whole sub-problems, usually with some output.

I find it much easier to read tests that call spec-documented APIs directly. Less to learn.

Others may disagree, but I'd rather see repetition in the tests themselves than too many helpers.